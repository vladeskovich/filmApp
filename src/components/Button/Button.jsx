import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Button.scss';

const Button = ({
  type,
  outline,
  color,
  className,
  children,
  ...props
}) => {
  const classes = classNames(
    styles.button,
    className,
    { [styles.outline]: outline },
    { [styles[color]]: color },
  );

  return (
    <button
      type={type}
      className={classes}
      {...props}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(['submit', 'button']),
  className: PropTypes.string,
  color: PropTypes.string,
  onClick: PropTypes.func,
  outline: PropTypes.bool,
};

Button.defaultProps = {
  type: 'button',
};

export default Button;
