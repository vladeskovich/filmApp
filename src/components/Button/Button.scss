.button {
  min-width: 130px;
  padding: 7px;
  color: white;
  outline: none;
  border-radius: 4px;
  font-size: 16px;
  background: rgba(0, 0, 0, 0.5);
  cursor: pointer;
}

.outline {
  border: 2px solid white;
}

.blue {
  border: 2px solid #0aaee4;
  background-color: #0aaee4;
}
