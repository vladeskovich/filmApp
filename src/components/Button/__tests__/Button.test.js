import React from 'react';
import Button from '../Button';

describe('Button', () => {
  const props = {
    type: 'submit',
    outline: true,
    color: 'blue',
    className: 'buttonTest',
  };

  it('should render Button component with props', () => {
    const component = shallow(<Button {...props}>Ok</Button>);

    expect(component).toMatchSnapshot();
  });
});
