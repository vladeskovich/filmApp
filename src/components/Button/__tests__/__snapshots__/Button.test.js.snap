// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button component with props 1`] = `
<button
  className="Button__button_1WSrj4ig buttonTest Button__outline_2AOV17gW Button__blue_2Th7Zabu"
  type="submit"
>
  Ok
</button>
`;
